### ISSUE TITLE
pg_transport Missing Documentation and pitfalls

### ISSUE THREAD

**Issue by Ilhicas** [awsdocs/amazon-rds-user-guide#95]
I'm currently running into an issue with pg_tranport that only outputs the following message: ERROR:  failed to download file data
SQL state: XX000

There is no documentation on errors that are outputted from the function, and given the closed source nature of the function, there is simply no way to understand what might be missing.

**Comment by miketwenty1** [comment]
I'm seeing this issues all over the place with running pg_transport multiple times. First time works. Consecutive times with other databases don't work. Must something going on because pg_transport doesn't work the second time unless I reboot the DB.

**Comment by baniol** [comment]
Also experiencing the issue. My workaround was to create a new migration database in target RDS for each source database and running SELECT there once.

**Comment by OzRgz** [comment]
I'm experiencing same issue, with a database around 330 GB always fails at 9-10% with 'failed to download file data' error. Any update?

**Comment by Matti-Saemi** [comment]
I am facing the same issue as well. I was hoping to use this extension. Will there be any fix for this anytime soon? or should I go back to using `pg_dump` :(?

**Comment by OswaldoRGZ** [comment]
Have anyone tried with this new Postgresql 12 on AWS RDS? my last atempt was with postgresql 10.x

**Comment by Ilhicas** [comment]
This was tried with a 11.5 version, haven't tried since, as we've resorted to using pg_dump and restore automated with good old ansible trying to mimic the behaviour by suspending write on the source to avoid inconsistencies.

**Comment by miketwenty1** [comment]
This [aws blog](https://aws.amazon.com/blogs/database/migrating-databases-using-rds-postgresql-transportable-databases/) some parameter suggestions for things like "num_workers" and "worker_mem". 
I noticed bumping resources for the DBs involved and setting parameters helps, but doesn't solve the issue completely.

**Comment by OswaldoRGZ** [comment]
Somehow its working now, tested on PostgreSQL 12.4.
400 GB in 20 mins with two m5.xlarge.

**Comment by randyurbano** [comment]
Thanks for the feedback. I'm closing this old issue. If changes are needed a the documentation in the RDS User Guide, re-open the issue and specify the topic and the changes that are needed.

**Comment by santiagortiiz** [comment]
Hi, I have encountered this problem recently.

What I notice is that the pg_transport.num_workers must be configured according to the database instance type.
I have been using Postgres version 14 on a db.t3.micro with 20GB of storage.

When I set pg_transport.num_workers > 3, I get the same error (Failed to download file data) at some point (50%, 30%).

So in the end I solve it by setting the workers to 2.
